docs/client-sdk/javascript/tutorials/build-an-xmtp-hello-world-app.md

```diff
@@ -0,0 +1,576 @@
+   // Wagmi client
+   const { provider } = configureChains(chains, [
+     walletConnectProvider({ projectId: "<YOUR_PROJECT_ID>" }),
+   ]);
+   const wagmiClient = createClient({
+     autoConnect: true,
+     connectors: modalConnectors({ appName: "web3Modal", chains }),
+     provider,
+   });
+   // Web3Modal Ethereum Client
+   const ethereumClient = new EthereumClient(wagmiClient, chains);
+
+   function App() {
+     // highlight-next-line
+     const { isConnected } = useAccount()
+
+     return (
+       <div className="App">
+         <Web3Button/>
+         <Web3Modal
+           projectId="<YOUR_PROJECT_ID>"
+           ethereumClient={ethereumClient}
+         />
+         // highlight-next-line
+        {isConnected && <button onClick={initXmtp}>Connect to XMTP</button>}
+       </div>
+     );
+   }
+
+   export default App;
+   ```
+   So now, the **Connect to XMTP** button should only show when your wallet is connected.
+   
+3. Now, let's add an onClick handler when users click the button. This will initialize XMTP with your app.
+   ```tsx title="App.tsx" showLineNumbers
+   import React from 'react';
+   import './App.css';
+   import {
+     EthereumClient,
+     modalConnectors,
+     walletConnectProvider,
+   } from "@web3modal/ethereum";
+   import { Web3Modal, Web3Button } from "@web3modal/react";
+   import { configureChains, createClient,} from "wagmi";
+   import { mainnet, polygon } from "wagmi/chains";
+   // highlight-start
+   import { fetchSigner } from '@wagmi/core'
+   import { Client } from '@xmtp/xmtp-js'
+   // highlight-end
+   
+   const chains = [ mainnet, polygon];
+   // Wagmi client
+   const { provider } = configureChains(chains, [
+     walletConnectProvider({ projectId: "<YOUR_PROJECT_ID>" }),
+   ]);
+   const wagmiClient = createClient({
+     autoConnect: true,
+     connectors: modalConnectors({ appName: "web3Modal", chains }),
+     provider,
+   });
+   // Web3Modal Ethereum Client
+   const ethereumClient = new EthereumClient(wagmiClient, chains);
+
+   function App() {
+     const { isConnected } = useAccount()
+     // highlight-start
+     const initXmtp = async function() {
+       const signer = await fetchSigner()
+       const xmtp = await Client.create(signer, { env: "production" })
+     }
+     // highlight-end
+     return (
+       <div className="App">
+         <Web3Button/>
+         <Web3Modal
+           projectId="<YOUR_PROJECT_ID>"
+           ethereumClient={ethereumClient}
+         />
+        {isConnected && <button onClick={initXmtp}>Connect to XMTP</button>}
+       </div>
+     );
+   }
+
+   export default App;
+   ```
+   At this point, clicking the **Connect to XMTP** button will ask your wallet for a signature request. Below is an example of a MetaMask Signature request screen for signing to enable an XMTP identity.
+
+   So now that we created the **Connect Wallet** and **Connect to XMTP** button, let's put it all together and display and send messages next!
```

Comment:
At this time, if I do `npm run start` - I get the following error. The error is not present when I finish building the app. 

I'm not sure if it is just me - but if this is expected, I wonder if it might be helpful to let the user know that they might see this error before they finish? Some users might get hung up on it and try to resolve it before moving on. =)

```
webpack compiled with 127 warnings
ERROR in src/App.tsx:28:27
TS2304: Cannot find name 'useAccount'.
    26 |
    27 | function App() {
  > 28 |   const { isConnected } = useAccount()
       |                           ^^^^^^^^^^
    29 |   const initXmtp = async function() {
    30 |     const signer = await fetchSigner()
    31 |     const xmtp = await Client.create(signer, { env: "production" })
```

Reply:
ahh, the `useAccount` exports is not in the code snippet. Thanks for catching. Added in the code and also highlighted in the code base